pkg/client/interfaces_test.go

```diff
@@ -32,6 +34,72 @@ func TestConfigValidation(t *testing.T) {
 		valid         bool
 		expectedError string
 	}{
+		{
+			desc: "gen config with valid yaml value is valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yaml": "test: \"valid\"",
+						},
+					},
+				},
+			},
+			valid: true,
+		},
+		{
+			desc: "gen config with valid yml value is valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yml": "test: \"valid\"",
+						},
+					},
+				},
+			},
+			valid: true,
+		},
+		{
+			desc: "gen config with invalid yaml value is not valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yaml": "test: \"in\"valid\"",
+						},
+					},
+				},
+			},
+			valid:         false,
+			expectedError: "failed to parse value of key test.yaml in ConfigMap for plugin test: error converting YAML to JSON: yaml: did not find expected key",
+		},
+		{
+			desc: "gen config with invalid yml value is not valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yml": "test: \"in\"valid\"",
```

Comment:
Totally fine to leave like this but a tip that I find helpful is that if you have a string with quotes, use the backtick instead of quotes. It will prevent `"` from being a special character and it doesnt have to be escaped.